### DISCUSSION TITLE
Viable soundfonts for alphatab

### DISCUSSION THREAD

**Discussion by MDoder97** [CoderLine/alphaTab#562] [Q&A] [answered]
Where can viable (1-2mb max) soundfonts be found for alphatab? Especially for electric since most of the good ones are way too large at 15mb+.

Can perhaps unnecessary sounds of those fonts be cut somehow in polyphone etc. to reduce it to a more manageable size?

Thanks.

**Comment by Danielku15** [comment, marked as answer]
I have following hints for you on this topic: 

1. You could try to make your own "optimized" soundfont with reduced size. You can slice it down to even load individual instruments only or some groups. This guide helps on this topic: https://alphatab.net/docs/guides/multiple-soundfonts 

2. Be sure to provide the file on your server with the right caching flags. This way users will only download this once. You could also store it manually in the local storage and load it form there if available. 

3. Sonivox is one of the best free soundfonts I could find. It is the soundfont used in Android and turned out to be the best fit for alphaTab. It's already very small but having a decent sound. 

4. I found this resource quite good on looking for soundfonts https://musical-artifacts.com/ 

AlphaTab currently only supports SoundFont2 which contains uncompressed PCM Samples. Having a full General Midi SoundFont with decent sound below 10MB you will hardly find, audio samples are simply so big. I have it still in mind to implement SoundFont3 for OGG/Vorbis encoded audio samples. This comes at cost at loading time, but reduces time/size on download. Long time ago I already had a PoC on this but I never brought it fully to life: https://twitter.com/alphaTabMusic/status/855864264250273792?s=20
